### PR TITLE
e2e: block all master addresses

### PR DESCRIPTION
**Pull request by justinsb**
This way we can be sure that the kubelet can't communicate with the
master, even if falls-back to the internal/external IP (which seems to
be the case with DNS)

Issue #56787

/kind failing-test

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -4934,19 +4934,28 @@ func getMaster(c clientset.Interface) Address {
...
-// which is the address of the interface used for communication with the kubelet.
-func GetMasterAddress(c clientset.Interface) string {
+// GetAllMasterAddresses returns all IP addresses on which the kubelet can reach the master.
+// It may return internal and external IPs, even if we expect for
+// e.g. internal IPs to be used (issue #56787), so that we can be
+// sure to block the master fully during tests.
+func GetAllMasterAddresses(c clientset.Interface) []string {
 	master := getMaster(c)
+
+	var ips sets.String
```
Comment (Huang-Wei) [inline test/e2e/framework/util.go]: +1 on using a set here - don't need extra logic to check if externalIP == internalIP.

**Comment by Huang-Wei** [comment]
It seems following logic (in test/e2e/network/firewall.go) panics:


```go
 		By("Checking if e2e firewall rules are correct")
 		for _, expFw := range gce.GetE2eFirewalls(cloudConfig.MasterName, cloudConfig.MasterTag, cloudConfig.NodeTag, cloudConfig.Network, cloudConfig.ClusterIPRange) {
 			fw, err := gceCloud.GetFirewall(expFw.Name)
 			Expect(err).NotTo(HaveOccurred())
 			Expect(gce.VerifyFirewallRule(fw, expFw, cloudConfig.Network, false)).NotTo(HaveOccurred())
 		}
```

**Comment by justinsb** [comment]
Finally these failures both look unrelated!

/retest

**Review by Huang-Wei — APPROVED** [review]
/lgtm

**Comment by justinsb** [comment]
Failure is a performance failure, and I don't think we're in that path

/retest

**Comment by Huang-Wei** [comment]
Kindly ping @bowei, could you please approve this PR?

**Pull request merged**